alterpartitionreassignments.go

```diff
@@ -0,0 +1,121 @@
+package kafka
+
+import (
+	"context"
+	"net"
+	"time"
+
+	"github.com/segmentio/kafka-go/protocol/alterpartitionreassignments"
+)
+
+// AlterPartitionReassignmentsRequest is a request to the AlterPartitionReassignments API.
+type AlterPartitionReassignmentsRequest struct {
+	// Address of the kafka broker to send the request to.
+	Addr net.Addr
+
+	// Topic is the name of the topic to alter partitions in.
+	Topic string
+
+	// Assignments is the list of partition reassignments to submit to the API.
+	Assignments []AlterPartitionReassignmentsRequestAssignment
+
+	// Timeout is the amount of time to wait for the request to complete.
+	Timeout time.Duration
+}
+
+// AlterPartitionReassignmentsRequestAssignment contains the requested reassignments for a single
+// partition.
+type AlterPartitionReassignmentsRequestAssignment struct {
+	// PartitionID is the ID of the partition to make the reassignments in.
+	PartitionID int
+
+	// BrokerIDs is a slice of brokers to set the partition replicas to.
+	BrokerIDs []int
+}
+
+// AlterPartitionReassignmentsResponse is a response from the AlterPartitionReassignments API.
+type AlterPartitionReassignmentsResponse struct {
+	// ErrorCode is set to a non-zero value if a top-level error was encountered.
+	ErrorCode int
+
+	// ErrorMessage describes the top-level error that occured.
+	ErrorMessage string
```

Comment:
We've expose errors as `error` types in other exported APIs of the client (for example https://pkg.go.dev/github.com/segmentio/kafka-go#FetchResponse), what do you think about continuing this pattern?

(and similarly in other types that include ErrorCode/ErrorMessage)

Reply:
Makes sense, done!